AEPCore/Sources/core/MobileCore.swift

```diff
@@ -41,17 +41,31 @@ public final class MobileCore: NSObject {
     /// - Parameter extensions: The extensions to be registered
     /// - Parameter completion: Closure to run when extensions have been registered
     @objc(registerExtensions:completion:)
-    public static func registerExtensions(_ extensions: [Extension.Type], _ completion: (() -> Void)? = nil) {
+    public static func registerExtensions(_ extensions: [NSObject.Type], _ completion: (() -> Void)? = nil) {
         let idParser = IDParser()
         V4Migrator(idParser: idParser).migrate() // before starting SDK, migrate from v4 if needed
         V5Migrator(idParser: idParser).migrate() // before starting SDK, migrate from v5 if needed
 
+        // Invoke registerExtension on legacy extensions
+        let legacyExtensions = extensions.filter {!($0.self is Extension.Type)} // All extensions that do not conform to `Extension`
+        let registerSelector = Selector(("registerExtension"))
+
+        for legacyExtension in legacyExtensions
+            where legacyExtension.responds(to: registerSelector) {
+            legacyExtension.perform(registerSelector)
+            if NSClassFromString("ACPBridgeExtension") == nil {
```

Comment:
Should we even perform the registerSelector if we don't have the compat layer installed?  Seems like it'd be better to put the if outside of the for loop and not even run the for loop if it's not there...